sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,183 @@
+        :type trim_prefixes: list[str]
+        :keyword key_vault_options: Options for resolving Key Vault references
+        :type key_vault_options: ~azure.appconfigurationprovider.KeyVaultOptions
+        """
+        provider = AzureAppConfigurationProvider()
+
+        key_vault_options = kwargs.pop("key_vault_options", None)
+
+        provider.__buildprovider(connection_string, endpoint,
+                                 credential, key_vault_options)
+
+        selects = kwargs.pop("selects", {SettingSelector("*", "\0")})
+        provider._trim_prefixes = kwargs.pop("trimmed_key_prefixes", [])
+
+        provider._dict = {}
+        secret_clients = {}
+
+        for select in selects:
+            configurations = provider._client.list_configuration_settings(
+                key_filter=select.key_filter, label_filter=select.label_filter)
+            for config in configurations:
+                if config.content_type is None:
+                    # Deals with possible null value via Rest API
+                    provider._dict[provider.__trim(config.key)] = config.value
+                elif config.content_type == "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8":
+                    provider.__resolve_keyvault_references(
+                        config, key_vault_options, secret_clients)
+                elif "application/json" in config.content_type:
+                    j_object = json.loads(config.value)
+                    provider._dict[provider.__trim(config.key)] = j_object
+                else:
+                    provider._dict[provider.__trim(config.key)] = config.value
+        return provider
+
+    def __buildprovider(self, connection_string, endpoint, credential, key_vault_options):
+        headers = {}
+        correlation_context = "RequestType=Startup"
+
+        if (key_vault_options is not None and
+                (key_vault_options.credential is not None or key_vault_options.secret_clients is not None or
+                 key_vault_options.secret_resolver is not None)):
+            correlation_context += ",UsesKeyVault"
+
+        headers["Correlation-Context"] = correlation_context
+        useragent = USER_AGENT
+
+        if connection_string is not None:
+            self._client = AzureAppConfigurationClient.from_connection_string(
+                connection_string, user_agent=useragent, headers=headers)
+            return
+        self._client = AzureAppConfigurationClient(
+            endpoint, credential, user_agent=useragent, headers=headers)
+
+    def __resolve_keyvault_references(self, config, key_vault_options, secret_clients):
+        if key_vault_options is None:
+            raise AttributeError(
+                "Key Vault options must be set to resolve Key Vault references.")
+        j_object = json.loads(config.value)
+        uri_value = j_object['uri']
+
+        uri = urlparse(uri_value)
```

Comment:
I don't think it does. It just throws anything not a uri into the fragment field. It will cause the logic bellow to throw an index out of bounds error for the path. But the rest should be fine.

Reply:
We should be prepared for any kind of value going into urlparse. There are cases where ValueError may be thrown ([reference](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse)).
"index out of bounds error" is also not a good situation to be in. If there is any error in the format of KVR uri, we can immediately throw an error like "Invalid Key Vault reference".

Reply:
No longer are issue. I figured out what I was doing wrong so we can used KeyVaultSecretIdentifier, which checks for us.